### ISSUE TITLE
Stuck on white screen when debugging iOS

### ISSUE THREAD

**Issue by jerryhamby** [flutter/flutter#63800]
I have been posting to the "Google Flutter Group" about this issue, here is the link.
https://groups.google.com/g/flutter-dev/c/YDzmNKd7LEA

Since moving my app to 1.20.0, 1.20.1, 1.20.2 this is a persistent problem.

When I attempt tp output my app to my IOS iPhone 11,  the installing process seems to work (the icon will show up) but the launching is where the problem happens. The screen goes white and stays white. It seems worse now on 1.20.2???

The only work around I have found is doing a "flutter clean" every time I want to output to my device. This is a pain.
Before 1.20.0, I didn't have any problems moving my app to my device.

My system:
Flutter 1.20.2
iPhone 11: 13.6.1
iMac:  10.15.6

<details>
 <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 1.20.2, on Mac OS X 10.15.6 19G2021, locale en-US)
    • Flutter version 1.20.2 at /Users/jerryhamby/Desktop/FlutterExamples/flutter
    • Framework revision bbfbf1770c (29 hours ago), 2020-08-13 08:33:09 -0700
    • Engine revision 9d5b21729f
    • Dart version 2.9.1

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/jerryhamby/Library/Android/sdk
    • Platform android-30, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.8.4

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio copy.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.48.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2

[✓] Connected device (1 available)
    • iPhone11 (mobile) • 00008030-000650610E40802E • ios • iOS 13.6.1
```
</details>

**Comment by mehmetalikaradeniz** [comment]
> I have been posting to the "Google Flutter Group" about this issue, here is the link.
> https://groups.google.com/g/flutter-dev/c/YDzmNKd7LEA
> 
> Since moving my app to 1.20.0, 1.20.1, 1.20.2 this is a persistent problem.
> 
> When I attempt tp output my app to my IOS iPhone 11, the installing process seems to work (the icon will show up) but the launching is where the problem happens. The screen goes white and stays white. It seems worse now on 1.20.2???
> 
> The only work around I have found is doing a "flutter clean" every time I want to output to my device. This is a pain.
> Before 1.20.0, I didn't have any problems moving my app to my device.
> 
> My system:
> Flutter 1.20.2
> iPhone 11: 13.6.1
> iMac: 10.15.6
> 
> [✓] Flutter (Channel stable, 1.20.2, on Mac OS X 10.15.6 19G2021, locale en-US)
> • Flutter version 1.20.2 at /Users/jerryhamby/Desktop/FlutterExamples/flutter
> • Framework revision [bbfbf17](https://github.com/flutter/flutter/commit/bbfbf1770cca2da7c82e887e4e4af910034800b6) (29 hours ago), 2020-08-13 08:33:09 -0700
> • Engine revision 9d5b21729f
> • Dart version 2.9.1
> 
> [✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
> • Android SDK at /Users/jerryhamby/Library/Android/sdk
> • Platform android-30, build-tools 29.0.2
> • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
> • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
> • All Android licenses accepted.
> 
> [✓] Xcode - develop for iOS and macOS (Xcode 11.6)
> • Xcode at /Applications/Xcode.app/Contents/Developer
> • Xcode 11.6, Build version 11E708
> • CocoaPods version 1.8.4
> 
> [✓] Android Studio (version 4.0)
> • Android Studio at /Applications/Android Studio copy.app/Contents
> • Flutter plugin version 47.1.2
> • Dart plugin version 193.7361
> • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
> 
> [✓] Android Studio (version 4.0)
> • Android Studio at /Applications/Android Studio.app/Contents
> • Flutter plugin version 47.1.2
> • Dart plugin version 193.7361
> • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
> 
> [✓] VS Code (version 1.48.0)
> • VS Code at /Applications/Visual Studio Code.app/Contents
> • Flutter extension version 3.13.2
> 
> [✓] Connected device (1 available)
> • iPhone11 (mobile) • 00008030-000650610E40802E • ios • iOS 13.6.1

Same Error Here

**Comment by harivamshi81189** [comment]
> I have been posting to the "Google Flutter Group" about this issue, here is the link.
> https://groups.google.com/g/flutter-dev/c/YDzmNKd7LEA
> 
> Since moving my app to 1.20.0, 1.20.1, 1.20.2 this is a persistent problem.
> 
> When I attempt tp output my app to my IOS iPhone 11, the installing process seems to work (the icon will show up) but the launching is where the problem happens. The screen goes white and stays white. It seems worse now on 1.20.2???
> 
> The only work around I have found is doing a "flutter clean" every time I want to output to my device. This is a pain.
> Before 1.20.0, I didn't have any problems moving my app to my device.
> 
> My system:
> Flutter 1.20.2
> iPhone 11: 13.6.1
> iMac: 10.15.6
> 
> [✓] Flutter (Channel stable, 1.20.2, on Mac OS X 10.15.6 19G2021, locale en-US)
> • Flutter version 1.20.2 at /Users/jerryhamby/Desktop/FlutterExamples/flutter
> • Framework revision [bbfbf17](https://github.com/flutter/flutter/commit/bbfbf1770cca2da7c82e887e4e4af910034800b6) (29 hours ago), 2020-08-13 08:33:09 -0700
> • Engine revision 9d5b21729f
> • Dart version 2.9.1
> 
> [✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
> • Android SDK at /Users/jerryhamby/Library/Android/sdk
> • Platform android-30, build-tools 29.0.2
> • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
> • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
> • All Android licenses accepted.
> 
> [✓] Xcode - develop for iOS and macOS (Xcode 11.6)
> • Xcode at /Applications/Xcode.app/Contents/Developer
> • Xcode 11.6, Build version 11E708
> • CocoaPods version 1.8.4
> 
> [✓] Android Studio (version 4.0)
> • Android Studio at /Applications/Android Studio copy.app/Contents
> • Flutter plugin version 47.1.2
> • Dart plugin version 193.7361
> • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
> 
> [✓] Android Studio (version 4.0)
> • Android Studio at /Applications/Android Studio.app/Contents
> • Flutter plugin version 47.1.2
> • Dart plugin version 193.7361
> • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
> 
> [✓] VS Code (version 1.48.0)
> • VS Code at /Applications/Visual Studio Code.app/Contents
> • Flutter extension version 3.13.2
> 
> [✓] Connected device (1 available)
> • iPhone11 (mobile) • 00008030-000650610E40802E • ios • iOS 13.6.1

same problem here too.

**Comment by markusaksli-nc** [comment]
Hi @jerryhamby 
What command are you using to "output the app"? If it is `flutter run` could you please provide your `flutter run --verbose logs`? Could you try to see if this behavior persists on the latest master?
Thank you

**Comment by jerryhamby** [comment]
I believe I'm issuing the latest stable IDE (1.20.2)
I dislike using the terminal, I mainly use the IDE (Visual Studio Code)

====================== Terminal VS CODE.ffutter run ====================
=============== This loaded to my device just fine ===================
Running pod install...                                              7.7s
Running Xcode build...                                                  
                                                   
 └─Compiling, linking and signing...                        15.3s
Xcode build done.                                           37.0s
Installing and launching...                                        31.7s
Waiting for iPhone11 to report its views...                          3ms
Syncing files to device iPhone11...                                316ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
……………………………..

====================== VS CODE. ====================
My next step was using VS Code (normally for me), I get the white screen:

Running pod install...
Xcode build done.                                           22.4s
Installing and launching...

**Comment by markusaksli-nc** [comment]
This is just a continuation of https://github.com/flutter/flutter/issues/22338, but that thread was locked anyway.

I can reproduce this with the following steps:

1. `flutter create bug`
2. Open Xcode, change bundle identifier and add signing
2.1 As a side note I don't use the "Open iOS module in Xcode" action in Android Studio since this runs `flutter build ios --simulator` in all cases for some reason.
3. Select physical device and run from Android Studio 4.0.1

This results in the app installing and opening but getting stuck on a white screen at "Syncing files to device". I can fix this for the project by running it from Xcode. Running from the IDE works after this. This can still sometimes pop back up when I've reconfigured the project during triage.
<details>
 <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 1.21.0-10.0.pre.136, on Mac OS X 10.15.6 19G73,
    locale en-GB)
    • Flutter version 1.21.0-10.0.pre.136 at
      /Users/markus/development/flutter_master
    • Framework revision 9e0ad9b6b5 (2 hours ago), 2020-08-18 01:36:02 -0400
    • Engine revision b9243713a0
    • Dart version 2.10.0 (build 2.10.0-36.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /Users/markus/Library/Android/sdk
    • Platform android-30, build-tools 30.0.1
    • Java binary at: /Users/markus/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/193.6626763/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Users/markus/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/193.6626763/Android
      Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

 
[✓] Connected device (5 available)            
    • Nevercode’s iPhone (mobile)         •
      b668e524315069f3db3661ac11ff1f66afafebdb • ios            • iOS 13.6
    • iPhone SE (2nd generation) (mobile) • E88D23AA-71AC-49A6-A418-058CDE0D2FC4
      • ios            • com.apple.CoreSimulator.SimRuntime.iOS-13-6 (simulator)
    • macOS (desktop)                     • macos
      • darwin-x64     • Mac OS X 10.15.6 19G73
    • Web Server (web)                    • web-server
      • web-javascript • Flutter Tools
    • Chrome (web)                        • chrome
      • web-javascript • Google Chrome 84.0.4147.125

• No issues found!
```
</details>

**Comment by mehmetalikaradeniz** [comment]
I am facing with this problem at Android Studio. 
I upgraded Flutter to version:  1.20.2. 
Physical Device is: IPhone 11 pro OS version 13.6
Simulator: IPhone 11 pro OS Version 13.6
Android Studio version: 4.0.1

Errors has started with this upgrade and  i face with this error on Physical Device and Simulator.
I cant build project without -- flutter clean command. If build succeed(sometimes it dont be succeed) it is stucking on "Syncing files to device" phase(on Just Physical device). After that i am terminating build section. Loop starts again for me  flutter clean and build again bla bla. This error amazing annoying

--> flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 1.20.2, on Mac OS X 10.15.6 19G73, locale en-TR)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
[✓] Android Studio (version 4.0)
[!] VS Code (version 1.47.3)
    ✗ Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
[✓] Connected device (2 available)

**Comment by harivamshi81189** [comment]
It's the issue with new flutter update.
FROM...
*D. Harivamshi*

**Comment by dim-apparence** [comment]
I'm facing exactly the same issue on iPhone X 13.6.1 with latest stable Flutter SDK/VSCode & Flutter extensions.
When I try to launch the app on debug mode from VSCode, white screen appear & stuck on "Syncing files to device".

It's work perfectly when launching command: ```flutter run``` from command line...

**Comment by jerryhamby** [comment]
I tried @markusaksli-nc advice about launching from Xcode and that seemed to work (for now).

I do hope the good follows at Flutter will fix this horrible workflow.

**Comment by jerryhamby** [comment]
FYI, this Xcode fix is veryyyy temporary. I'm back to square 1.

**Comment by mehmetalikaradeniz** [comment]
Xcode fix is not working. i tried many times and i got different errors.

an example --> 
Command PhaseScriptExecution failed with a nonzero exit code

**Comment by markusaksli-nc** [comment]
@jerryhamby @mehmetalikaradeniz
The discussion in the previously mentioned issue has some other solutions for this. As a general rule I won't recommend `flutter clean` or deleting any config files since that will obfuscate the issue here.

Please see https://github.com/flutter/flutter/issues/22338#issuecomment-542629985 on how to add useful logs to this issue. We need actionable information to get to understand why you are running into this and what we can improve.

Cc @DanTup

**Comment by jerryhamby** [comment]
This output is from adding "args": ["-v"] to the launch.json file. Maybe it will help???
I think this starts just before the white page happens.
Haven't had time to do a log file yet.

Why is this happening in 2020 when it was discovered in 2018?  (#22338)

+321 ms] [ 52%] CreatingStagingDirectory
[        ] [ 57%] ExtractingPackage
[        ] [ 60%] InspectingPackage
[  +11 ms] [ 60%] TakingInstallLock
[   +2 ms] [ 65%] PreflightingApplication
[ +325 ms] [ 65%] InstallingEmbeddedProfile
[  +14 ms] [ 70%] VerifyingApplication
[ +865 ms] [ 75%] CreatingContainer
[  +14 ms] [ 80%] InstallingApplication
[   +5 ms] [ 85%] PostflightingApplication
[   +3 ms] [ 90%] SandboxingApplication
[  +13 ms] [ 95%] GeneratingApplicationMap
[  +81 ms] [100%] Installed package build/ios/iphoneos/Runner.app
[ +152 ms] ------ Debug phase ------
[        ] Starting debug of 00008030-000650610E40802E (N104AP, N104AP, uknownos, unkarch) a.k.a. 'iPhone11' connected through USB...
[+2967 ms] [  0%] Looking up developer disk image
[  +14 ms] [ 95%] Developer disk image mounted successfully
[+6017 ms] [100%] Connecting to remote debug server
[        ] -------------------------
[  +24 ms] (lldb) command source -s 0 '/tmp/3A5ADEB6-85BB-4275-A81B-8BEF4BB19411/fruitstrap-lldb-prep-cmds-00008030_000650610E40802E'
[        ] Executing commands in '/tmp/3A5ADEB6-85BB-4275-A81B-8BEF4BB19411/fruitstrap-lldb-prep-cmds-00008030_000650610E40802E'.
[        ] (lldb)     platform select remote-ios --sysroot '/Users/jerryhamby/Library/Developer/Xcode/iOS DeviceSupport/13.6.1 (17G80) arm64e/Symbols'
[        ]   Platform: remote-ios
[        ]  Connected: no
[        ]   SDK Path: "/Users/jerryhamby/Library/Developer/Xcode/iOS DeviceSupport/13.6.1 (17G80) arm64e/Symbols"
[        ] (lldb)     target create "/Users/jerryhamby/Desktop/FlutterExamples/vdex/build/ios/iphoneos/Runner.app"
[+5077 ms] Current executable set to '/Users/jerryhamby/Desktop/FlutterExamples/vdex/build/ios/iphoneos/Runner.app' (arm64).
[        ] (lldb)     script fruitstrap_device_app="/private/var/containers/Bundle/Application/A5006EDA-DC09-4BAF-84C3-CF578393287A/Runner.app"
[ +153 ms] (lldb)     script fruitstrap_connect_url="connect://127.0.0.1:57133"
[        ] (lldb)     script fruitstrap_output_path=""
[        ] (lldb)     script fruitstrap_error_path=""
[        ] (lldb)     target modules search-paths add /usr "/Users/jerryhamby/Library/Developer/Xcode/iOS DeviceSupport/13.6.1 (17G80) arm64e/Symbols/usr" /System "/Users/jerryhamby/Library/Developer/Xcode/iOS DeviceSupport/13.6.1 (17G80) arm64e/Symbols/System" "/private/var/containers/Bundle/Application/A5006EDA-DC09-4BAF-84C3-CF578393287A" "/Users/jerryhamby/Desktop/FlutterExamples/vdex/build/ios/iphoneos" "/var/containers/Bundle/Application/A5006EDA-DC09-4BAF-84C3-CF578393287A" "/Users/jerryhamby/Desktop/FlutterExamples/vdex/build/ios/iphoneos" /Developer "/Users/jerryhamby/Library/Developer/Xcode/iOS DeviceSupport/13.6.1 (17G80) arm64e/Symbols/Developer"
[  +27 ms] (lldb)     command script import "/tmp/3A5ADEB6-85BB-4275-A81B-8BEF4BB19411/fruitstrap_00008030_000650610E40802E.py"
[   +2 ms] (lldb)     command script add -f fruitstrap_00008030_000650610E40802E.connect_command connect
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_00008030_000650610E40802E.run_command run
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_00008030_000650610E40802E.autoexit_command autoexit
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_00008030_000650610E40802E.safequit_command safequit
[        ] (lldb)     connect
[  +31 ms] (lldb)     run
[ +424 ms] success
[        ] (lldb)     safequit
[ +108 ms] Process 410 detached
[  +42 ms] Application launched on the device. Waiting for observatory port.
[  +10 ms] Attempting to forward device port 50406 to host port 57142
[        ] executing: /Users/jerryhamby/Desktop/FlutterExamples/flutter/bin/cache/artifacts/usbmuxd/iproxy 57142:50406 --udid 00008030-000650610E40802E
[+1008 ms] Forwarded port ForwardedPort HOST:57142 to DEVICE:50406
[  +11 ms] HttpException: Connection closed before full header was received, uri = http://localhost:57142/ws
[+2141 ms] Caching compiled dill
[  +41 ms] Connecting to service protocol: http://localhost:57142
[  +21 ms] Successfully connected to service protocol: http://localhost:57142
[  +10 ms] DevFS: Creating new filesystem on the device (null)
[  +11 ms] DevFS: Created new filesystem on the device (file:///private/var/mobile/Containers/Data/Application/443E4D65-952C-4613-8E83-865AE8298F67/tmp/vdexRRpQh6/vdex/)
[   +1 ms] Updating assets
[ +256 ms] Manifest contained wildcard assets. Inserting missing file into build graph to force rerun. for more information see #56466.
[   +7 ms] Scanning asset files
[   +4 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[   +1 ms] <- recompile package:vdex/main.dart 90e316c0-9f4e-4448-9f9b-f4703c75644d
[        ] <- 90e316c0-9f4e-4448-9f9b-f4703c75644d
[  +57 ms] Updating files
Exited

**Comment by jerryhamby** [comment]
I waited almost 3 minutes (guesstimate) and the app finally loaded to the iPhone 11 device.
It's a very large file and I have the whole thing if you really need it.
This log info is basically where I think the white page started.

+303 ms] [ 52%] CreatingStagingDirectory
[        ] [ 57%] ExtractingPackage
[        ] [ 60%] InspectingPackage
[   +9 ms] [ 60%] TakingInstallLock
[   +2 ms] [ 65%] PreflightingApplication
[ +157 ms] [ 65%] InstallingEmbeddedProfile
[   +6 ms] [ 70%] VerifyingApplication
[ +321 ms] [ 75%] CreatingContainer
[   +7 ms] [ 80%] InstallingApplication
[   +5 ms] [ 85%] PostflightingApplication
[   +4 ms] [ 90%] SandboxingApplication
[   +8 ms] [ 95%] GeneratingApplicationMap
[  +37 ms] [100%] Installed package build/ios/iphoneos/Runner.app
[ +177 ms] ------ Debug phase ------
[        ] Starting debug of 00008030-000650610E40802E (N104AP, N104AP, uknownos, unkarch) a.k.a. 'iPhone11' connected through USB...
[+3037 ms] [  0%] Looking up developer disk image
[  +15 ms] [ 95%] Developer disk image mounted successfully
[+6090 ms] [100%] Connecting to remote debug server
[        ] -------------------------
[  +24 ms] (lldb) command source -s 0 '/tmp/E92034FB-F87A-4F94-B5CF-3681DCAA6CF0/fruitstrap-lldb-prep-cmds-00008030_000650610E40802E'
[        ] Executing commands in '/tmp/E92034FB-F87A-4F94-B5CF-3681DCAA6CF0/fruitstrap-lldb-prep-cmds-00008030_000650610E40802E'.
[        ] (lldb)     platform select remote-ios --sysroot '/Users/jerryhamby/Library/Developer/Xcode/iOS DeviceSupport/13.6.1 (17G80) arm64e/Symbols'
[        ]   Platform: remote-ios
[        ]  Connected: no
[        ]   SDK Path: "/Users/jerryhamby/Library/Developer/Xcode/iOS DeviceSupport/13.6.1 (17G80) arm64e/Symbols"
[        ] (lldb)     target create "/Users/jerryhamby/Desktop/FlutterExamples/vdex/build/ios/iphoneos/Runner.app"
[+5139 ms] Current executable set to '/Users/jerryhamby/Desktop/FlutterExamples/vdex/build/ios/iphoneos/Runner.app' (arm64).
[        ] (lldb)     script fruitstrap_device_app="/private/var/containers/Bundle/Application/DF6756C1-B9B1-45B8-B4BA-3DAE77E8F97B/Runner.app"
[ +173 ms] (lldb)     script fruitstrap_connect_url="connect://127.0.0.1:57896"
[        ] (lldb)     script fruitstrap_output_path=""
[        ] (lldb)     script fruitstrap_error_path=""
[        ] (lldb)     target modules search-paths add /usr "/Users/jerryhamby/Library/Developer/Xcode/iOS DeviceSupport/13.6.1 (17G80) arm64e/Symbols/usr" /System "/Users/jerryhamby/Library/Developer/Xcode/iOS DeviceSupport/13.6.1 (17G80) arm64e/Symbols/System" "/private/var/containers/Bundle/Application/DF6756C1-B9B1-45B8-B4BA-3DAE77E8F97B" "/Users/jerryhamby/Desktop/FlutterExamples/vdex/build/ios/iphoneos" "/var/containers/Bundle/Application/DF6756C1-B9B1-45B8-B4BA-3DAE77E8F97B" "/Users/jerryhamby/Desktop/FlutterExamples/vdex/build/ios/iphoneos" /Developer "/Users/jerryhamby/Library/Developer/Xcode/iOS DeviceSupport/13.6.1 (17G80) arm64e/Symbols/Developer"
[  +28 ms] (lldb)     command script import "/tmp/E92034FB-F87A-4F94-B5CF-3681DCAA6CF0/fruitstrap_00008030_000650610E40802E.py"
[   +2 ms] (lldb)     command script add -f fruitstrap_00008030_000650610E40802E.connect_command connect
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_00008030_000650610E40802E.run_command run
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_00008030_000650610E40802E.autoexit_command autoexit
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_00008030_000650610E40802E.safequit_command safequit
[        ] (lldb)     connect
[  +26 ms] (lldb)     run
[ +421 ms] success
[        ] (lldb)     safequit
[ +109 ms] Process 430 detached
[  +38 ms] Application launched on the device. Waiting for observatory port.
[   +9 ms] Attempting to forward device port 50406 to host port 57905
[        ] executing: /Users/jerryhamby/Desktop/FlutterExamples/flutter/bin/cache/artifacts/usbmuxd/iproxy 57905:50406 --udid 00008030-000650610E40802E
[+1009 ms] Forwarded port ForwardedPort HOST:57905 to DEVICE:50406
[ +248 ms] Caching compiled dill
[  +36 ms] Connecting to service protocol: http://localhost:57905
[  +16 ms] Successfully connected to service protocol: http://localhost:57905
[   +9 ms] DevFS: Creating new filesystem on the device (null)
[   +8 ms] DevFS: Created new filesystem on the device (file:///private/var/mobile/Containers/Data/Application/BCCCAC58-9D5F-492B-B23F-E104E60D6B9B/tmp/vdexvpKubA/vdex/)
[   +2 ms] Updating assets
[ +265 ms] Manifest contained wildcard assets. Inserting missing file into build graph to force rerun. for more information see #56466.
[   +8 ms] Scanning asset files
[   +4 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[        ] <- recompile package:vdex/main.dart 8b5002fd-12b8-4c0a-a68c-fc03b937f312
[        ] <- 8b5002fd-12b8-4c0a-a68c-fc03b937f312
[  +57 ms] Updating files
[+239328 ms] Error writing "build/flutter_assets/assets/add.png" to DevFS: HttpException: , uri = http://localhost:57905
[        ] trying again in a few - 9 more attempts left
[ +519 ms] DevFS: Sync finished
[   +1 ms] Synced 2.0MB.
[        ] <- accept
[   +3 ms] Connected to _flutterView/0x10a83ee20.
Connecting to VM Service at ws://localhost:57905/ws
flutter: Splash initState stated ...
flutter: clearThePrefs started = accountid
flutter: setDrawerSettingsOff started ...
flutter: MAIN setUpDetails theos = ios
flutter: Splash initState ended ...
flutter: Splash initState version = 1.0.0
flutter: Splash initState buildNumber = 1
flutter: %%%%% getSuper started...........
flutter: %%%%% provider started ...........
flutter: SuperProvider finalobjSuper = [{appinfostatus: on, commentstatus: on, sharestatus: on, dicid: super_abcOGQWRx12345, vdexstatus: on, accountid: abcOGQWRx12345, partnerstatus: on, url: https://www.vdex.com}]
flutter: getSuper ended...........
flutter: MyVars.deviceheight = 896.0
flutter: navigationToNextPage deviceheight = 896.0
flutter: 496.0
flutter: 400.0
flutter: MyVars.deviceheight = 896.0
flutter: entering LoginPage ...
[+10549 ms] DevFS: Deleting filesystem on the device (file:///private/var/mobile/Containers/Data/Application/BCCCAC58-........-E104E60D6B9B/tmp/vdexvpKubA/vdex/)
[  +41 ms] DevFS: Deleted filesystem on the device (file:///private/var/mobile/Containers/Data/Application/BCCCAC58-9D5F-492B-B23F-E104E60D6B9B/tmp/vdexvpKubA/vdex/)
[  +46 ms] Service protocol connection closed.
Application finished.
Exited (sigterm)
[   +4 ms] "flutter run" took 313,602ms.
[ +169 ms] ensureAnalyticsSent: 166ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hook priority 4
[        ] Shutdown hooks complete
[   +1 ms] exiting with code 0

**Comment by Jay-Zhang** [comment]
I met this issue too when developing a plugin package, the reason for me is that the iOS code of my plugin crashed in initState method of statefulWidget，then I stucked on white screen.  If I  Open iOS module in Xcode and launch from Xcode，the console will show the crash log clearly. When I fix the crash in iOS code, the app didn't stuck on white screen anymore.

**Comment by TahaTesser** [comment]
This got worse, the white screen doesn't go away even after running `flutter run` of a couple of times. Complete kill it and tried it, it works on the third or fourth  try  on latest master

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel master, 1.22.0-2.0.pre.36, on Mac OS X 10.15.6 19G2021,
    locale en-GB)
    • Flutter version 1.22.0-2.0.pre.36 at /Users/tahatesser/Code/flutter_master
    • Framework revision d30e36ba8c (2 days ago), 2020-08-21 22:36:03 -0400
    • Engine revision d495da20d0
    • Dart version 2.10.0 (build 2.10.0-49.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.1
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.48.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2

 
[✓] Connected device (4 available)            
    • iPhone 11 (mobile) • A63D6F51-59F3-4A75-961F-3769709F3A92 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-13-6 (simulator)
    • macOS (desktop)    • macos                                • darwin-x64
      • Mac OS X 10.15.6 19G2021
    • Web Server (web)   • web-server                           • web-javascript
      • Flutter Tools
    • Chrome (web)       • chrome                               • web-javascript
      • Google Chrome 84.0.4147.135

• No issues found!
```

</details>

**Comment by jerryhamby** [comment]
I have been going to Flutter and selecting "open in Xcode", then launching my app to my device from Xcode. It's a pain and I hope someone comes up with a fix.  So far this workaround is working every time. This is for debug mode only, haven't tested for run mode yet.

**Comment by jmagman** [comment]
I've been trying on master and a new project `flutter run`s with Xcode 11.6 installed without any white screen, I've launched it like 10 times now.  

"White screen" means an engine crash.  If someone reproducing this can open Console, select their phone, and see what logs are happening from their app while they are reproducing, that would be helpful.

**Comment by jerryhamby** [comment]
Thanks for testing on the master platform, I only run on the stable platform.
I just tried a very small app and didn't have any problems going to my iPhone 11 device (white screen).

The big but here is I'm still having problems with the white screen on my very large app,
(65 pages, 23 classes and 29 dependency plugins in the yaml file]

Again, this white screen issue was never an problem in Flutter versions below 1.20

**Comment by fnicastri** [comment]
I confirm the behavior.
Sometimes if you wait a couple of minutes it starts, sometimes it just exit.
see this for my logs: [#63025-comment](https://github.com/flutter/flutter/issues/63025#issuecomment-678615444)

**Comment by xvrh** [comment]
I see the same issue: iPhone is stuck for +200 seconds with "Syncing files to device".
It appears very randomly.
Log shows:
```
[  +72 ms] Updating files
[+222192 ms] Error writing "build/flutter_assets/assets/images/3x/login_bg.png" to DevFS: SocketException: OS Error: Connection reset by peer, errno = 54, address = 127.0.0.1, port = 51004
[        ] trying again in a few - 9 more attempts left
[ +537 ms] DevFS: Sync finished
[        ] Synced 17.4MB.
[        ] <- accept
```

**Comment by bobmoff** [comment]
Same as @xvrh but with a slightly different error message.

```
[  +76 ms] Updating files
[+238755 ms] Error writing "build/flutter_assets/packages/cupertino_icons/assets/CupertinoIcons.ttf" to DevFS: HttpException: , uri = http://localhost:65230
[        ] trying again in a few - 9 more attempts left
[        ] Error writing "build/flutter_assets/AssetManifest.json" to DevFS: HttpException: , uri = http://localhost:65230
[        ] trying again in a few - 9 more attempts left
[ +533 ms] DevFS: Sync finished
```

The "Run with XCode"-trick usually works, but it always comes back again. Have been trying all day to find a pattern to this behavior but no luck yet.

**Comment by TahaTesser** [comment]
Hi @jmagman 

Here what I have found

**VS Code**
When the app is running from VS Code, It is often stuck at a white screen for 5-15 min, it says `syncing files to device`. Sometimes it loads the app after a long wait or lost connection. Sometimes even after multiple tries, it is white screen  Seems to be related to how many packages are in `pubspec.yaml`, longer it takes.   This used to be 1-2 min but recently it got worse with the latest dev and master. This doesn't happen every run, only at a first-run after flutter clean or newly added packages 

![IMG20200825210905](https://user-images.githubusercontent.com/48603081/91855395-be402580-ec82-11ea-8b91-0574423a1c62.jpg)
![IMG20200825210856](https://user-images.githubusercontent.com/48603081/91855398-bed8bc00-ec82-11ea-8a32-57c5ce919da7.jpg)

**Android Studio**
I don't use it much for flutter, just for Android projects, so I can't comment my experience with it but it does load fine with Android Studio 

 
**XCode** 
Run apps fine, there is no white screen  or  crash 

**Terminal**
Run apps fine, there is no white screen  or crash 


<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel master, 1.22.0-2.0.pre.162, on Mac OS X 10.15.6 19G2021,
    locale en-GB)
    • Flutter version 1.22.0-2.0.pre.162 at
      /Users/tahatesser/Code/flutter_master
    • Framework revision bcfad2068e (77 minutes ago), 2020-09-01 08:34:25 -0400
    • Engine revision 165abef0a2
    • Dart version 2.10.0 (build 2.10.0-77.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.48.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2

[✓] Connected device (4 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E • ios            • iOS 13.6.1
    • macOS (desktop)        • macos                     • darwin-x64     • Mac OS X 10.15.6
      19G2021
    • Web Server (web)       • web-server                • web-javascript • Flutter Tools
    • Chrome (web)           • chrome                    • web-javascript • Google Chrome
      84.0.4147.135

• No issues found!
```

</details>

**Comment by fnicastri** [comment]
@TahaTesser 
On Android Studio I get exactly the same behavior.
You can see my logs at this [comment](https://github.com/flutter/flutter/issues/63025#issuecomment-678615444)

**Comment by fnicastri** [comment]
@TahaTesser 
from a quick test the situation gets much better if you add every single asset file to pubspec.

**Comment by w3ggy** [comment]
Are there any possible solutions? Idk why I can't use debugger on my iOS device eve after `flutter clean`. It is freezing on syncing files step.

**Comment by bobmoff** [comment]
I usually try the following, in this order. 

If 1 doesnt work on its own I try again but with first 2 AND then 1. If that doesnt work i have to do all of the three, in reversed order, and after that there is usually not any syncing issues. 

1. Build (just build, no need to also run) using XCode
2. run `flutter clean`
3. remove app from device

When I have gotten VSCode connected and working on the device and try my hardest to not do anything that will make it disconnect. I turned off the sleep lock completely.

Good luck to everyone out there! 🤓

**Comment by akyjoe** [comment]
When I get stuck on white screen or native splash screen: 

1. I disconnect the USB Cable from the Mac and reinsert again.
2. I disconnect the Lightening Jack from the iDevice and reinsert again.
3. Rerun the App from Android Studio again.

There is definitely a link drop from either the iDevice or the Mac due to a hardware or software bug. Apple devices are well known for their closed systems and perhaps some security protocol is flagged up. Another thing to notice is iOS 13.6 had issues while "Trusting the Computer", whenever the above steps were followed, it prompted to "Trust the Computer". After upgrading to 13.7, it is reduced.

Possible factors:
- Could be a weak voltage drop or a data link break/delay while charging of the iDevice from computer.
- Link break because of Device inactivity or Sleep Lock as mentioned by @bobmoff .
- Faulty Data link cable
- Faulty USB Port

Issue is definitely there, but it can be narrowed down by the @fluttergithubbot principals.

**Comment by wonpyohong** [comment]
i confirmed that 
flutter v1.20.3 -> sometimes stuck. succeed after 2-5 repetition.
flutter v1.17.5 -> always succeed

by changing version with command 'flutter version v1.xx.x'

If it is due to hardware(jack) or OS version, I cannot figure out this

**Comment by matpag** [comment]
I have the same problem while running on iOS devices, the app starts only after 4 minutes of white screen. 
This was not happening in 1.17 version of Flutter
<details>
<summary>flutter run logs</summary>

```
....
[  +74 ms] Application launched on the device. Waiting for observatory port.
[  +11 ms] Attempting to forward device port 52262 to host port 64520
[        ] executing: /Users/$USER/Library/Flutter/bin/cache/artifacts/usbmuxd/iproxy 64520:52262 --udid 00008027-000D18D01E11002E
[+1039 ms] Forwarded port ForwardedPort HOST:64520 to DEVICE:52262
[ +119 ms] Caching compiled dill
[  +48 ms] Connecting to service protocol: http://localhost:64520
[  +24 ms] Successfully connected to service protocol: http://localhost:64520
Waiting for iPad Pro 11 of $USER to report its views...
[   +9 ms] DevFS: Creating new filesystem on the device (null)
[  +11 ms] DevFS: Created new filesystem on the device (file:///private/var/mobile/Containers/Data/Application/2FC68A2D-F50F-4597-B548-C4A183AA77DB/tmp/appzfA8ui/app/)
[   +1 ms] Updating assets
Debug service listening on ws://localhost:64520/ws
[ +175 ms] Manifest contained wildcard assets. Inserting missing file into build graph to force rerun. for more information see #56466.
Syncing files to device iPad Pro 11 of $USER...
[   +9 ms] Scanning asset files
[   +1 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[        ] <- recompile package:$APP/main_dev.dart c71e8fb3-9fca-440f-a4b8-6938677a62a0
[        ] <- c71e8fb3-9fca-440f-a4b8-6938677a62a0
[  +56 ms] Updating files
[+239284 ms] Error writing "build/flutter_assets/assets/icons/.DS_Store" to DevFS: HttpException: , uri = http://localhost:64520
[        ] trying again in a few - 9 more attempts left
[        ] Error writing "build/flutter_assets/packages/cupertino_icons/assets/CupertinoIcons.ttf" to DevFS: HttpException: , uri = http://localhost:64520
[        ] trying again in a few - 9 more attempts left
[ +544 ms] DevFS: Sync finished
[   +2 ms] Synced 1.4MB.
[   +1 ms] <- accept
[   +3 ms] Connected to _flutterView/0x11a825a20.
```

</details>

**Comment by jerryhamby** [comment]
???
Same problem showing on 1.20.4, I don't see any improvements.
How about some positive feedback? Sure makes debugging a pain...

**Comment by mwawrusch** [comment]
Having this problem for months now, and now started to see a new problem which might be related. It compiles great, shortly shows in the simulator, then aborts. here is that log file:

```
[+2500 ms] executing: /usr/bin/xcrun simctl install 627483D0-B33B-4F1B-91B5-4FFE52696DF0 /Users/mwawrusch/Dev/roji/flutter/roji_wallet/build/ios/iphonesimulator/Runner.app
[+2575 ms] executing: /usr/bin/plutil -convert json -o - /Users/mwawrusch/Dev/roji/flutter/roji_wallet/build/ios/iphonesimulator/Runner.app/Info.plist
[   +9 ms] Exit code 0 from: /usr/bin/plutil -convert json -o - /Users/mwawrusch/Dev/roji/flutter/roji_wallet/build/ios/iphonesimulator/Runner.app/Info.plist
[        ]
{"DTCompiler":"com.apple.compilers.llvm.clang.1_0","CFBundleInfoDictionaryVersion":"6.0","UISupportedInterfaceOrientations~ipad":["UIInterfaceOrientationPortrait","UIInterfaceOrientationPortraitUpsideDown","UIInterfaceOrientatio
nLandscapeLeft","UIInterfaceOrientationLandscapeRight"],"DTPlatformVersion":"13.7","com.claimsforce.segment.ENABLE_AMPLITUDE_INTEGRATION":false,"CFBundleName":"roji_wallet","DTSDKName":"iphonesimulator13.7","NSPhotoLibraryUsageD
escription":"Used to select a profile
picture.","UIViewControllerBasedStatusBarAppearance":false,"com.claimsforce.segment.TRACK_APPLICATION_LIFECYCLE_EVENTS":false,"CFBundleIcons":{"CFBundlePrimaryIcon":{"CFBundleIconFiles":["AppIcon60x60"],"CFBundleIconName":"AppIc
on"}},"NSLocationAlwaysAndWhenInUseUsageDescription":"Used to ensure the authenticity of your account and to prevent fraud.","CFBundleDisplayName":"Roji","LSRequiresIPhoneOS":true,"NSLocationWhenInUseUsageDescription":"Used to
ensure the authenticity of your account and to prevent fraud.","UIBackgroundModes":["remote-notification"],"DTSDKBuild":"17H22","NSCameraUsageDescription":"Used to take a profile
picture.","CFBundleShortVersionString":"1.0.0","CFBundleSupportedPlatforms":["iPhoneSimulator"],"NSLocationAlwaysUsageDescription":"Used to ensure the authenticity of your account and to prevent
fraud.","UISupportedInterfaceOrientations":["UIInterfaceOrientationPortrait","UIInterfaceOrientationLandscapeLeft","UIInterfaceOrientationLandscapeRight"],"BuildMachineOSBuild":"19G2021","DTPlatformBuild":"17H22","CFBundlePackag
eType":"APPL","DTXcodeBuild":"11E801a","CFBundleDevelopmentRegion":"en","MinimumOSVersion":"12.4","CFBundleVersion":"3","NSFileProviderDomainUsageDescription":"Used to take a profile
picture.","io.flutter.embedded_views_preview":true,"com.claimsforce.segment.WRITE_KEY":"5QiCT1q451MJIT5cCHlj3gAPtkJlzrsZ","UIDeviceFamily":[1,2],"UILaunchStoryboardName":"LaunchScreen","CFBundleIdentifier":"io.roji.wallet","DTXc
ode":"1170","NSLocationUsageDescription":"Used to ensure the authenticity of your account and to prevent
fraud.","CFBundleExecutable":"Runner","CFBundleIcons~ipad":{"CFBundlePrimaryIcon":{"CFBundleIconFiles":["AppIcon60x60","AppIcon76x76"],"CFBundleIconName":"AppIcon"}},"CFBundleSignature":"????","NSFaceIDUsageDescription":"Used to
provide an additional layer of
security.","DTPlatformName":"iphonesimulator","UIMainStoryboardFile":"Main","CFBundleURLTypes":[{"CFBundleTypeRole":"Editor","CFBundleURLSchemes":["com.googleusercontent.apps.203255661000-q0fqvgobvqoobpgjn3pvq36cj1fdds3i"]}]}
[  +15 ms] executing: /usr/bin/xcrun simctl launch 627483D0-B33B-4F1B-91B5-4FFE52696DF0 io.roji.wallet --enable-dart-profiling --enable-checked-mode --verify-entry-points --observatory-port=0
[ +705 ms] io.roji.wallet: 7296
[        ] Waiting for observatory port to be available...
[+4059 ms] Observatory URL on device: http://127.0.0.1:65438/5Tcumae0MVE=/
[   +3 ms] Caching compiled dill
[+4273 ms] Connecting to service protocol: http://127.0.0.1:65438/5Tcumae0MVE=/
[ +125 ms] Successfully connected to service protocol: http://127.0.0.1:65438/5Tcumae0MVE=/
[        ] Waiting for iPhone 11 to report its views...
[   +4 ms] Waiting for iPhone 11 to report its views... (completed in 3ms)
[   +7 ms] DevFS: Creating new filesystem on the device (null)
[  +14 ms] DevFS: Created new filesystem on the device
(file:///Users/mwawrusch/Library/Developer/CoreSimulator/Devices/627483D0-B33B-4F1B-91B5-4FFE52696DF0/data/Containers/Data/Application/CE097917-C96D-44E8-89DB-6C415C93D79D/tmp/roji_walletcf71ej/roji_wallet/)
[   +2 ms] Updating assets
[+2911 ms] Syncing files to device iPhone 11...
[   +1 ms] Scanning asset files
[  +31 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[   +1 ms] <- recompile package:roji_wallet/main.dart a8fd4398-a204-4db3-96ee-53a396a16648
[        ] <- a8fd4398-a204-4db3-96ee-53a396a16648
[   +1 ms] [Crashlytics] Version 3.14.0 (144)
[        ] [Fabric] [Fabric +with] called multiple times. Only the first call is honored, please pass all kits you wish to initialize
[        ] [Crashlytics:Crash:Reports] Packaged report with id '17e07ac3c9524cb891fd1cc991c69d33' for submission
[ +112 ms] Updating files
[ +277 ms] flutter:
eyJraWQiOiJoYjlNa3lCVW1LdDBESWxsSnIwbU9pVTNCdEhmMlI0UENQRmFTRnEyXC9VWT0iLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiIwMDc0NGIzNS0zMzkzLTQxMzEtODY3Ni1hMTVmYjUwNDEwNDciLCJldmVudF9pZCI6IjMwZjExNTNlLWIzZmEtNGMxNS05MjAzLWI3NTMzOWVmNWM4NiIsInRva2V
uX3VzZSI6ImFjY2VzcyIsInNjb3BlIjoiYXdzLmNvZ25pdG8uc2lnbmluLnVzZXIuYWRtaW4iLCJhdXRoX3RpbWUiOjE2MDAwOTE5NzUsImlzcyI6Imh0dHBzOlwvXC9jb2duaXRvLWlkcC51cy1lYXN0LTEuYW1hem9uYXdzLmNvbVwvdXMtZWFzdC0xX29UbFJxOWVvOSIsImV4cCI6MTYwMDIzODk3NCw
iaWF0IjoxNjAwMjM1Mzc0LCJqdGkiOiI1YTExY2QyNS04MWZmLTRmMjctODdiNi0xOGU0NWU2ODE4MjEiLCJjbGllbnRfaWQiOiI2b3BpOG1waDdlZm1zZDM2bDRjNWc5czkxbSIsInVzZXJuYW1lIjoiMjY3NTdhZmU5ODlmNGE4MzgxYzFkMzQ4NWEyOTc0ODkifQ.Fnl_ESG1xVFjZ_3w3UnDlYO9Qnr7
VjAOr9HQ0Ynv1huEUZaBqP4AgUBICDhWwGxKkyyUY7aZwPqs-zZaJZpVhUbLyR1Nvaz70gN56ZFnraYO4_a4IhI7q2lq-n7hC59vedS8WJ-TDrE9TgT1dnYXbAWnEEQ0ymq2HbKdpYnFv6Hu2CwlrC-GyeooAw5dmGId_UFuK4oLslKR7-dQoQSI8dNwbXfvDaUHCWuvxjZlXWrlZC_HeT9EsqCvs5aYkr1N
TrQnMypAZCnGZVAgpde87K2sp09dzf7SDKM7gByrsAOybP0txRhoXl9tSLv3_ZC6F0nZ_EhvZz3MqmSjT3Zkjg
[+1457 ms] Error writing "build/flutter_assets/packages/intl_phone_number_input/assets/flags/ky.png" to DevFS: HttpException: Connection closed before full header was received, uri = http://127.0.0.1:65438/5Tcumae0MVE=/
[        ] trying again in a few - 9 more attempts left
[  +49 ms] Service protocol connection closed.
[        ] Lost connection to device.
[   +5 ms] Error writing "build/flutter_assets/packages/intl_phone_number_input/assets/flags/gn.png" to DevFS: SocketException: Write failed (OS Error: Broken pipe, errno = 32), address = 127.0.0.1, port = 65465
[        ] trying again in a few - 9 more attempts left
[        ] Error writing "build/flutter_assets/packages/intl_phone_number_input/assets/flags/is.png" to DevFS: SocketException: OS Error: Connection reset by peer, errno = 54, address = 127.0.0.1, port = 65467
[        ] trying again in a few - 9 more attempts left
[        ] Error writing "build/flutter_assets/packages/intl_phone_number_input/assets/flags/eg.png" to DevFS: SocketException: OS Error: Connection reset by peer, errno = 54, address = 127.0.0.1, port = 65468
[        ] trying again in a few - 9 more attempts left
[        ] Error writing "build/flutter_assets/packages/intl_phone_number_input/assets/flags/fk.png" to DevFS: SocketException: OS Error: Connection reset by peer, errno = 54, address = 127.0.0.1, port = 65469
[        ] trying again in a few - 9 more attempts left
[        ] Error writing "build/flutter_assets/packages/intl_phone_number_input/assets/flags/fj.png" to DevFS: SocketException: Write failed (OS Error: Broken pipe, errno = 32), address = 127.0.0.1, port = 65466
[        ] trying again in a few - 9 more attempts left
[ +448 ms] Error writing "build/flutter_assets/packages/intl_phone_number_input/assets/flags/ky.png" to DevFS: SocketException: OS Error: Connection refused, errno = 61, address = 127.0.0.1, port = 65474
[        ] trying again in a few - 8 more attempts left
[  +54 ms] Error writing "build/flutter_assets/packages/intl_phone_number_input/assets/flags/gn.png" to DevFS: SocketException: OS Error: Connection refused, errno = 61, address = 127.0.0.1, port = 65475
[        ] trying again in a few - 8 more attempts left
[        ] Error writing "build/flutter_assets/packages/intl_phone_number_input/assets/flags/is.png" to DevFS: SocketException: OS Error: Connection refused, errno = 61, address = 127.0.0.1, port = 65476
[        ] trying again in a few - 8 more attempts left
[        ] Error writing "build/flutter_assets/packages/intl_phone_number_input/assets/flags/eg.png" to DevFS: SocketException: OS Error: Connection refused, errno = 61, address = 127.0.0.1, port = 65477
[        ] trying again in a few - 8 more attempts left
[        ] Error writing "build/flutter_assets/packages/intl_phone_number_input/assets/flags/fk.png" to DevFS: SocketException: OS Error: Connection refused, errno = 61, address = 127.0.0.1, port = 65478
[        ] trying again in a few - 8 more attempts left
[        ] Error writing "build/flutter_assets/packages/intl_phone_number_input/assets/flags/fj.png" to DevFS: SocketException: OS Error: Connection refused, errno = 61, address = 127.0.0.1, port = 65479
[        ] trying again in a few - 8 more attempts left
```

**Comment by mwawrusch** [comment]
I doubt it's hardware. First, it happens rarely on the simulator as well, secondly running from xcode always works.

**Comment by akyjoe** [comment]
> I doubt it's hardware. First, it happens rarely on the simulator as well, secondly running from xcode always works.

Okay, I agree it works with xCode in first run. So, if it's not the hardware then may be there's a broken link in between Android Studio and a Physical iDevice. Conclusively, either it's the Apple thing or the Google thing.

**Comment by gksrb4** [comment]
any update? 
It is miserable waiting 5 minute to run flutter or hot reload in high possibility.

**Comment by jmagman** [comment]
If you're stuck at "Syncing files to device" (@markusaksli-nc @mehmetalikaradeniz, @dim-apparence, @xvrh, @bobmoff) this should be alleviated with https://github.com/flutter/flutter/pull/66152, which is available on the [master channel](https://github.com/flutter/flutter/wiki/Flutter-build-release-channels#how-to-change-channels), and was just [hotfixed to the beta channel](https://github.com/flutter/flutter/pull/66849) (it's kind of weird but it hasn't made it to dev yet, but it will be there soon).

With @jerryhamby's verbose output, I can see it's the same thing:
```
[+239328 ms] Error writing "build/flutter_assets/assets/add.png" to DevFS: HttpException: , uri = http://localhost:57905
```

Duplicate of #65402.

**Comment by stx** [comment]
@jmagman Any ETA on this reaching stable? It's a significant workflow impediment for many of us shipping apps from stable.

**Comment by jmagman** [comment]
@stx Very soon.

**Comment by mnc12004** [comment]
Is this fixed in 1.22?

Using an iPhone 11 device, the app hangs at the white screen for up to an hour!

I've run all the above suggestions, even deleted the ios folder and recreated (flutter create .).

I've just downgraded to 1.22 hoping this fixes my issue.

**Comment by bobmoff** [comment]
I am on 1.22.3 (stable channel) and I dont have this issues any longer.

**Comment by developeranshul1986** [comment]
I am building the app in Flutter Stable Version 1.22.4 and running the app in simulator iOS 13.6 and white screen stick issue persist.

Please help on this as it needed urgent

**Comment by bobmoff** [comment]
I just created a new project with Flutter 1.22.4 and launched it on iPhone 12 Pro Max iOS 14.2 Simulator, and it works fine. Also tried on iPhone 5s iOS 12.2. 

It also works on device. I have an iPhone 11 Pro Max and there is no problem launching. Hot reload also works.

**Comment by developeranshul1986** [comment]
Please try to reproduce as per below
https://github.com/flutter/flutter/issues/63869

**Comment by hkmsadek** [comment]
Anyone solved? I am getting same error. After running `flutter run`, app shows a white screen and then crashed with the error `connection lost to device` :( 

My flutter version is 

```
Framework • revision 1aafb3a8b9 (4 weeks ago) • 2020-11-13 09:59:28 -0800
Engine • revision 2c956a31c0
Tools • Dart 2.10.4
```

Anyone ?
